locator/topology.cc

```diff
@@ -128,6 +128,15 @@ const endpoint_dc_rack& topology::get_location(const inet_address& ep) const {
     on_internal_error(tlogger, format("Node {} is not in topology", ep));
 }
 
+std::unordered_set<sstring> topology::get_datacenters() const {
+    std::unordered_set<sstring> ret;
+    ret.reserve(_dc_endpoints.size());
```

Comment:
This is an over-reservation. A cluster with 80 nodes but just 1 datacenter doesn't need to reserve 80 slots (and this will make iteration slower since we'll need to skip over empty buckets).

Reply:
btw, this is a good candidate for C++23 std::flat_set, when the time comes.

Reply:
> This is an over-reservation. A cluster with 80 nodes but just 1 datacenter doesn't need to reserve 80 slots (and this will make iteration slower since we'll need to skip over empty buckets).

`_dc_endpoints` is a `std::unordered_map<sstring, std::unordered_set<endpoint>>` so its size is the number of DCs.

Reply:
ah, right